src/main/java/com/exasol/containers/ExasolContainer.java

```diff
@@ -626,7 +629,45 @@ public void purgeDatabase() {
                 final Statement statement = connection.createStatement()) {
             new ExasolDatabaseCleaner(statement).cleanDatabase();
         } catch (final SQLException exception) {
-            throw new ExasolContainerInitializationException("Failed to purge database", exception);
+            throw new ExasolContainerInitializationException(
+                    ExaError.messageBuilder("E-ETC-24").message("Failed to purge database").toString(), exception);
+        }
+    }
+
+    /**
+     * Clean up BucketFS by deleting all objects from the default bucket.
+     */
+    // [impl->dsn~bucketfs-purging~1]
+    public void purgeDefaultBucket() {
+        final Bucket bucket = getDefaultBucket();
+        try {
+            bucket.listContentsRecursively().stream() //
+                    .filter(not(this::keep)) //
+                    .forEach(path -> deleteBucketFsPath(bucket, path));
+        } catch (final BucketAccessException exception) {
+            throw new ExasolContainerInitializationException(
+                    ExaError.messageBuilder("E-ETC-25").message("Failed to purge the default bucket").toString(),
+                    exception);
+        }
+    }
+
+    /**
+     * Ensure not to delete objects like
+     * {@code EXAClusterOS/ScriptLanguages-standard-EXASOL-7.1.0-slc-v4.0.0-CM4RWW6R.tar.gz}.
+     */
+    private boolean keep(final String bucketFsPath) {
+        return bucketFsPath.isEmpty() || bucketFsPath.startsWith("EXAClusterOS/");
```

Comment:
Should appear in the requirements!

Reply:
The `isEmpty()` breaks the separation of concerns, since it contradicts the word `keep`.

Invert the logic and call it `doesBucketFsObjectRequireDeletion`.